kitsune/search/tests/test_api.py

```diff
@@ -65,6 +67,30 @@ def test_invalid_product(self):
         eq_(res.status_code, 400)
         eq_(res.data['detail'], {'product': 'Could not find product with slug "nonexistant".'})
 
+    def test_invalid_locale(self):
+        res = self.client.get(reverse('search.suggest'), {
+            'locale': 'bad-medicine',
+            'q': 'search',
+        })
+        eq_(res.status_code, 400)
+        eq_(res.data['detail'], {'locale': 'Could not find locale "bad-medicine".'})
+
+    def test_invalid_fallback_locale(self):
+        for locale, fallback in settings.NON_SUPPORTED_LOCALES.items():
+            if fallback is not None:
+                break
```

Comment:
I don't think it is a misfeature, since "for" doesn't introduce a new scope in Python, nor is it supposed to. I'm not opposed to replacing this with something simpler though.